api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java

```diff
@@ -94,6 +98,26 @@ void outputMoreThanInputTest() {
     assertEquals("The transaction you are trying to build has more outputs than inputs", exception.getError().getMessage());
   }
 
+  @Test
+  void calculateTxSize_whenUnsignedTransactionAddressesInvalid_thenThrowException() {
```

Comment:
Do we have only one unit test for the `calculateTxSize()` method?

Reply:
Yeah, our services unit test coverage is rather poor